modules/Bio/EnsEMBL/Compara/PipeConfig/UpdateMSA_conf.pm

```diff
@@ -0,0 +1,437 @@
+                'column_names'  => [ 'table' ],
+            },
+            -flow_into  => {
+                '2->A' => [ 'copy_ancestral_table' ],
+                'A->1' => [ 'update_ancestor_names' ],
+            },
+        },
+        {   -logic_name => 'copy_ancestral_table',
+            -module     => 'Bio::EnsEMBL::Hive::RunnableDB::MySQLTransfer',
+            -parameters => {
+                'src_db_conn'   => $self->o('prev_ancestral_db'),
+                'dest_db_conn'  => $self->o('ancestral_db'),
+            },
+        },
+        {   -logic_name => 'update_ancestor_names',
+            -module     => 'Bio::EnsEMBL::Hive::RunnableDB::SqlCmd',
+            -parameters => {
+                'db_conn' => $self->o('ancestral_db'),
+                'sql'     => [
+                    'UPDATE seq_region SET name = REPLACE(name, "_#prev_mlss_id#_", "_#mlss_id#_")',
+                ],
+            },
+        },
+        # Transfer MLSS, GA* and dnafrag ids from the previous MLSS id to the current one, update the MSA and
+        # ancestral core database, and finally remove the previous MSA information
+        {   -logic_name => 'transfer_msa',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::UpdateMSA::TransferAlignment',
+            -parameters => {
+                'method_type' => $self->o('method_type'),
+            },
+            -flow_into  => [ 'affected_gabs_factory' ],
+        },
+        {   -logic_name => 'affected_gabs_factory',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::UpdateMSA::UpdateGABFactory',
+            -flow_into  => {
+                '2->A' => [ 'update_gab' ],
+                'A->1' => WHEN(
+                    '#method_type# eq "epo"' => [ 'sync_ancestral_database' ],
+                    ELSE                        [ 'remove_prev_mlss' ],
+                ),
+            },
+        },
+        {   -logic_name     => 'update_gab',
+            -module         => 'Bio::EnsEMBL::Compara::RunnableDB::UpdateMSA::UpdateGAB',
+            -rc_name        => '1Gb_job',
+            -batch_size     => 50,
+            -hive_capacity  => 250,
+            -flow_into      => {
+                2   => '?accu_name=ancestor_names&accu_address=[]&accu_input_variable=name',
+            }
+        },
+        {   -logic_name => 'sync_ancestral_database',
+            -module     => 'Bio::EnsEMBL::Compara::RunnableDB::UpdateMSA::SyncAncestralDB',
+            -parameters => {
+                'ancestral_db' => $self->o('ancestral_db'),
+            },
+            -flow_into  => [ 'remove_prev_mlss' ],
+        },
+        {   -logic_name => 'remove_prev_mlss',
+            -module     => 'Bio::EnsEMBL::Hive::RunnableDB::SqlCmd',
+            -parameters => {
+                'wrap_in_transaction' => 1,
+                'sql'                 => [
+                    'SET FOREIGN_KEY_CHECKS=0',
```

Comment:
There's no order to do these `DELETE`s in that doesn't require FKs to be turned off?

Reply:
I have checked and I didn't know you can actually tell `DELETE` to do its job in a specific order (we need to remove `node_id` in decreasing order so it doesn't conflict with `parent_id` in `species_tree_node`).